lib/Server.js

```diff
@@ -87,6 +87,11 @@ class Server {
       throw new Error("'filename' option must be set in lazy mode.");
     }
 
+    if (options.spdy || options.http2) {
+      options.spdy = true;
+      options.http2 = true;
+    }
```

Comment:
Let's separate `spdy` and `http2` options, because it is misleading, `spdy` and `https2` are difference protocols, better don't mix their

Reply:
@evilebottnawi I considered `http2` an alias for `spdy` because `http2` needs `spdy` to work at the moment. As an alternative, I could do this, to allow for future separation of `http2` from `spdy`: 

When ONLY `http2` is enabled, the spdy server will be run (since this is the only way to run HTTP/2 on express at the moment), and these protocols will be provided to spdy:

```js
options.https.spdy = {
  protocols: ['h2', 'http/1.1'],
};
```
Then the exact same thing will happen as above if `http2` is enabled and `spdy` is enabled.

If ONLY `spdy` is enabled, the `options.https.spdy` will not be set unless the user specifies it themselves. If the user does not specify it in their options, it will go to the default spdy value as specified [here](https://www.npmjs.com/package/spdy):

```js
protocols: ['h2','spdy/3.1', 'spdy/3', 'spdy/2','http/1.1', 'http/1.0']
```

Do you like this alternative better?

Reply:
Ok, let's use the `http2` option instead `spdy`. And we can introduce `spdy` under other option in future because `spdy` !== `http2`, also we need throw error if developer doesn't defined `https` when use `http2: true`

Reply:
Ok, that makes sense. Should user still be allowed to provide spdy options like this?:

```js
{
  contentBase: './dist',
  watchContentBase: true,
  https: {
    spdy: {
      protocols: ['h2', 'http/1.1' ...]
    }
  },
  http2: true
}
```

Reply:
@Loonride i thought about it, let's do this in other PR

Reply:
@evilebottnawi i think it would help if the `http2` option was nested under `https` because it makes it more obvious that http2 requires https. Perhaps even when only

```js
{
    https: {
        http2: true,
    },
}
```

Treat that as if both https (with self-signed cert) & http2 are enabled?

Reply:
hm, http2 !== https2, but no one browsers don't support http2 without https, but for avoid misleading better contain 2 option (i think), new developers can start think what http2 === https, but i am open for feedback

Reply:
Actually, I like [your suggestion below](#discussion_r267269542): assume `https: true` when `http2: true` and no https config is specified (and maybe output something in verbose logging about the assumption).

Reply:
Oh, duh. That's exactly what this code is doing 🤦‍♂️